app/src/main/java/net/aiscope/gdd_app/ui/sample_completion/SampleCompletionActivity.kt

```diff
@@ -105,52 +120,75 @@ class SampleCompletionActivity : CaptureFlow, AppCompatActivity() {
                 erroneousTab = 2
             }
             return erroneousTab
+    }
+
+    fun getCurrentTab(): Int
+    {
+        with(binding) {
+            return viewPager.currentItem
         }
     }
 
-    fun save() {
-        val erroneousTab = validateTabsAndUpdateVM();
-        if (erroneousTab == null) {
+    fun saveToVM() {
             try {
                 sharedVM.save()
-                finishFlow()
+                goToHomeAndConfirmSaved()
             } catch (@Suppress("TooGenericExceptionCaught") error: Throwable) {
                 Timber.e(error, "An error occurred when saving sample completion data")
-                showRetryBar()
+                //showRetryBar()
+                // Fn Moved here to temporary fix TooManyFunctions ¯\_(ツ)_/¯

```

Comment:
I mean, if we are just moving stuff cause of the 'toomanyfunctions' I'd rather we just suppress the rule

Reply:
I will move them back and add the suppress